test/smoke/testApps/OpenTelemetryMetric/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/OpenTelemetryMetricTest.java

```diff
@@ -0,0 +1,255 @@
+
+    Envelope rdEnvelope = rdList.get(0);
+    RequestData rd = (RequestData) ((Data<?>) rdEnvelope.getData()).getBaseData();
+    assertEquals("GET /OpenTelemetryMetric/" + name, rd.getName());
+
+    Envelope envelope = metrics.get(0);
+
+    // validate tags
+    Map<String, String> tags = envelope.getTags();
+    assertNotNull(tags.get("ai.internal.sdkVersion"));
+    assertEquals(tags.get("ai.cloud.roleInstance"), "testroleinstance");
+    assertEquals(tags.get("ai.cloud.role"), "testrolename");
+    assertEquals(tags.get("ai.application.ver"), "123");
+
+    // validate base data
+    MetricData md = (MetricData) ((Data<?>) envelope.getData()).getBaseData();
+    List<DataPoint> dataPointList = md.getMetrics();
+    assertEquals(1, dataPointList.size());
+    DataPoint dp = dataPointList.get(0);
+    double expectedValue = 10;
+    double epsilon = Math.ulp(expectedValue);
+    assertEquals(expectedValue, dp.getValue(), epsilon);
+    assertEquals(DataPointType.Measurement, dp.getKind());
+    assertEquals(name, dp.getName());
+
+    // validate custom dimension
+    Map<String, String> properties = md.getProperties();
+    assertEquals(properties.get("_MS.AggregationIntervalMs"), "60000");
+    assertEquals(properties.get("tag1"), "abc");
+    assertEquals(properties.get("tag2"), "def");
+    assertEquals(properties.get("thing1"), "thing2");
+  }
+
+  private void validateCounterMetric(String name) throws Exception {
+    List<Envelope> rdList = mockedIngestion.waitForItems("RequestData", 1);
+    List<Envelope> metrics = mockedIngestion.waitForItems(getMetricPredicate(name), 3, 40, TimeUnit.SECONDS);
+    assertEquals(3, metrics.size());
+
+    metrics.sort(Comparator.comparing(
+        obj -> {
+          MetricData metricData = (MetricData) ((Data<?>) obj.getData()).getBaseData();
+          List<DataPoint> dataPointList = metricData.getMetrics();
+          DataPoint dataPoint = dataPointList.get(0);
+          return dataPoint.getValue();
+        }));
+
+    Envelope rdEnvelope = rdList.get(0);
+    RequestData rd = (RequestData) ((Data<?>) rdEnvelope.getData()).getBaseData();
+    assertEquals("GET /OpenTelemetryMetric/" + name, rd.getName());
+
+    // validate 1st metric
+    Envelope envelope1 = metrics.get(0);
+
+    // validate tags
+    Map<String, String> tags = envelope1.getTags();
+    assertNotNull(tags.get("ai.internal.sdkVersion"));
+    assertEquals(tags.get("ai.cloud.roleInstance"), "testroleinstance");
+    assertEquals(tags.get("ai.cloud.role"), "testrolename");
+    assertEquals(tags.get("ai.application.ver"), "123");
+
+    // validate base data
+    MetricData md = (MetricData) ((Data<?>) envelope1.getData()).getBaseData();
+    List<DataPoint> dataPointList = md.getMetrics();
+    assertEquals(1, dataPointList.size());
+    DataPoint dp = dataPointList.get(0);
+    double expectedValue = 2.0;
+    double epsilon = Math.ulp(expectedValue);
+    assertEquals(expectedValue, dp.getValue(), epsilon);
```

Comment:
I don't think the usage in these tests can have floating point issues, I'd probably go with

```suggestion
    assertEquals(expectedValue, dp.getValue(), 0);
```

(I did similar here in MicrometerTest: https://github.com/microsoft/ApplicationInsights-Java/blob/0c054711d73d803d2e45c58d24216df09cab20bf/test/smoke/testApps/Micrometer/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/MicrometerTest.java#L74)